src/features/snackbar/index.js

```diff
@@ -117,8 +125,18 @@ class Snackbar extends Component {
         let showType = show ? show.substring(0, show.indexOf(':')) : show;
 
         return (
-            <div
+            <Dialog
                 className="snackbar__container white-border"
+                keys={[E_KEY]}
+                onKeyPress={() => {
+                    if (
+                        this.state.show.includes('NEW ITEM') &&
+                        this.state.item.type !== 'potion' &&
+                        this.state.item.name !== 'Rusty Sword'
+                    ) {
+                        this.handleEquip();
+                    }
+                }}
```

Comment:
This is a rather good start, and I agree we could probably tidy this up a little (as well as make it hopefully more performant).

My thinking is that perhaps we make this a part of the state of this component -- so we set the fact that it's something we can equip in the case that it's a 'NEW ITEM' notification, and update the state as necessary inside the 'componentDidUpdate' method. Let me know what you think, and if you want any guidance relating to it.